dev/webpack.config.js

```diff
@@ -0,0 +1,161 @@
+/* eslint import/no-commonjs: 0 */
+const path = require('path');
+const without = require('lodash/without');
+
+const webpack = require('webpack');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const HtmlWebpackPluging = require('html-webpack-plugin');
+const BabiliPlugin = require('babili-webpack-plugin');
+const AutoDllPlugin = require('autodll-webpack-plugin');
+const HappyPack = require('happypack');
+
+const __DEV__ = process.env.NODE_ENV === 'development';
+const clean = arr => arr.filter(item => item !== false);
+
+module.exports = {
```

Comment:
Why not do something in between where you split into different categories:

```js
const bundle = __DEV__ ? ...
const rules = [...];
if (__DEV__) {rules.push()...}
const plugins = [...];
if (__DEV__) plugins.push(autoreload);

module.exports = {...};
```

Not ideal but might ease the reading. I was also in favor for production/dev configs but reading this one, it's not that hard to read it (I am used to webpack still)